sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveTableValuedFunctions.scala

```diff
@@ -78,37 +80,52 @@ object ResolveTableValuedFunctions extends Rule[LogicalPlan] {
   private val builtinFunctions: Map[String, TVF] = Map(
     "range" -> Map(
       /* range(end) */
-      tvf("end" -> LongType) { case Seq(end: Long) =>
-        Range(0, end, 1, None)
+      tvf("end" -> LongType) { case (tvf, args @ Seq(end: Long)) =>
+        validateInputDimension(tvf, 1)
+        Range(0, end, 1, None, tvf.outputNames.headOption)
```

Comment:
great suggestion! I could make it simpler (https://github.com/apache/spark/pull/17666/files#diff-e2023817ad13201adeb45b74e99d29f8R129). Thought?